### PR TITLE
Removed Parse_Context include from Date class (MLDB-1215)

### PR DESCRIPTION
**Pull request by jeremybarnes**
Another simple change: value_description.h and parse_context.h were removed from the #includes for the Date class, and everything that broke was fixed.  Decouples the interfaces from the implementation and reduces compile times.

### CODE REVIEW

File: plugins/lang/python/callback.h [hunk 1]
```diff
@@ -1,9 +1,9 @@
-// This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
-
 /** callback.h                                 -*- C++ -*-
     Rémi Attab, 16 Jan 2013
     Copyright (c) 2013 Datacratic.  All rights reserved.
 
+    This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
```
Comment (FinchPowers) [inline plugins/lang/python/callback.h]: On a 2 Datacratic copyright maintenant. 2013 et 2015. C'est bon ça?

File: vfs/filter_streams.h [hunk 2]
```diff
@@ -1,9 +1,9 @@
-// This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
-
 /* filter_streams.h                                                -*- C++ -*-
    Jeremy Barnes, 12 March 2005
    Copyright (c) 2005 Jeremy Barnes.  All rights reserved.
    
+   This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
```
Comment (FinchPowers) [inline vfs/filter_streams.h]: 2005 vs 2015

File: vfs/fs_utils.h [hunk 3]
```diff
@@ -1,9 +1,9 @@
-// This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
-
 /* fs_utils.h                                                       -*- C++ -*-
    Wolfgang Sourdeau, February 2014
    Copyright (c) 2014 Datacratic.  All rights reserved.
 
+   This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
```
Comment (FinchPowers) [inline vfs/fs_utils.h]: 2014 vs 2015

File: vfs/filter_streams.h [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
+   
    This file is part of "Jeremy's Machine Learning Library", copyright (c)
    1999-2015 Jeremy Barnes.
```
Comment (FinchPowers) [inline vfs/filter_streams.h]: 1999! :p

File: plugins/lang/python/callback.h (hunk 1 above)
Comment (jeremybarnes) [inline plugins/lang/python/callback.h]: Yes... that's OK.

**Comment by FinchPowers** [comment]
+1

**Comment by jeremybarnes** [comment]
The copyrights are fine.  If you write a file in one year and modify it another, you have part of it copyrighted in each of the years.  Anyway, those changes just move comments around.  Any objections to the material changes?

**Pull request merged**